### PR TITLE
WIP: Windows support via binary download

### PR DESCRIPTION
**Pull request by richardotis**
Due to compiler licensing and technical difficulties with building all the dependencies ( https://github.com/conda-forge/mumps-feedstock/pull/17 https://github.com/conda-forge/scotch-feedstock/pull/13 ), I am suggesting we implement download and installation of an older, officially pre-built Ipopt version for Windows. As we resolve dependency issues, a Windows source build can be reinstated as normal and the Windows version bumped.

Linux and Mac would continue to build from source as normal.

Inspired by this conda recipe: https://anaconda.org/cachemeorg/ipopt_bin/3.11.1/download/win-64/ipopt_bin-3.11.1-5.tar.bz2

### CODE REVIEW

**Comment by richardotis** [comment]
cc @moorepants @bocklund

**Comment by richardotis** [comment]
It looks like a recipe maintainer will need to enable building on appveyor before the build can be tested.

**Comment by richardotis** [comment]
Also @jakirkham may want to chime in as to whether this would be an acceptable approach.

**Comment by pstjohn** [comment]
> It looks like a recipe maintainer will need to enable building on appveyor before the build can be tested.

How might one do this? 😄

**Comment by richardotis** [comment]
I think I can actually do it in the PR; let me check.

**Comment by jakirkham** [comment]
We are on `conda-smithy` 2.4.0 FYI. Requires `conda` 4.3 and `conda-build-all` 1.1.1.

**Comment by jakirkham** [comment]
> Also @jakirkham may want to chime in as to whether this would be an acceptable approach.

So we normally frown on repackaging binaries. That said, this is tricky case for sure given it requires Fortran. It may be possible to use the MSYS2 packages that @mingwandroid has put together to build this. Perhaps he can advise further.

**Comment by richardotis** [comment]
Thoughts on merging this? I'm having trouble testing the installed library files but the ipopt executable seems to work fine.

**Comment by pstjohn** [comment]
I'm pretty agnostic on it, how would it work having multiple versions of the package in the same repository? 

Is this needed as a dependency for another package? Or just as an alternative installation method for IPOPT specifically?

**Comment by richardotis** [comment]
As far as I know, there is no issue with multiple library versions in the same repo as long as the preprocessing selectors are set up to have the right version for each platform.

My interest in this is I maintain a downstream library which supports Windows, Mac and Linux, and IPOPT will be a dependency in our next major release. Having this package work well on conda will simplify the installation procedure for users on Windows.

**Comment by richardotis** [comment]
COIN-OR and AMPL ( http://ampl.com/products/solvers/open-source/#ipopt ) seem to have no problem distributing Windows binaries with just the EPL license included.

**Comment by bocklund** [comment]
Is anything blocking this @pstjohn @bluescarni?

**Comment by jakirkham** [comment]
FYI this is now failing on Linux `master`.

**Comment by moorepants** [comment]
I also suggest that the Intel license is included with this distribution if it isn't. It is also not noted in the README of this feedstock.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -5,40 +5,45 @@
...
-  sha256: {{ sha256 }}
+  fn: {{ name }}-{{ version }}.tar.gz  # [not win]
+  url: http://www.coin-or.org/download/pkgsource/{{ name }}/{{ name }}-{{ version }}.tgz  # [not win]
+  sha256: {{ sha256 }}  # [not win]
   patches:
-    - ipopt.patch
-
+    - ipopt.patch  # [not win]
+  fn: Ipopt-3.11.1-win64-intel13.1.zip  # [win64]
+  url: https://www.coin-or.org/download/binary/Ipopt/Ipopt-3.11.1-win64-intel13.1.zip  # [win64]
```
Comment (moorepants) [inline recipe/meta.yaml]: These lines should use the `{{ version }}` variable.

**Comment by richardotis** [comment]
Please revert this for now, I'm working an alternative using m2w64-toolchain which may allow us to do source builds.

On Nov 18, 2017 11:29 AM, jakirkham <notifications@github.com> wrote:

FYI this is now failing on Linux master.

**Comment by moorepants** [comment]
This is a bit problematic that master now has two versions of IPOPT in one recipe. All should be 3.12.8 or we should revert to 3.11.1 if that is the only available windows one.

**Pull request merged**